drivers/rtc/rtc_rv8263.c

```diff
@@ -422,89 +422,59 @@ static int rv8263c8_alarm_set_time(const struct device *dev, uint16_t id, uint16
 		return -EINVAL;
 	}
 
-	/* Disable the alarm when mask is zero. */
-	if (mask == 0) {
-		return i2c_reg_update_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_CONTROL_2,
-					      RV8263C8_BM_ALARM_INT_ENABLE,
-					      RV8263C8_BM_ALARM_INT_DISABLE);
-	}
-
 	if (!rtc_utils_validate_rtc_time(timeptr, mask)) {
 		LOG_ERR("Invalid mask!");
 		return -EINVAL;
 	}
 
-	regs[0] = bin2bcd(timeptr->tm_sec) & SECONDS_BITS;
-	regs[1] = bin2bcd(timeptr->tm_min) & MINUTES_BITS;
-	regs[2] = bin2bcd(timeptr->tm_hour) & HOURS_BITS;
-	regs[3] = bin2bcd(timeptr->tm_mday) & DATE_BITS;
-	regs[4] = bin2bcd(timeptr->tm_wday) & WEEKDAY_BITS;
-
-	if (mask & RTC_ALARM_TIME_MASK_SECOND) {
-		err = i2c_reg_write_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_SECONDS_ALARM,
-					    RV8263C8_BM_ALARM_ENABLE | regs[0]);
+	if (mask == 0) {
+		err = i2c_reg_update_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_CONTROL_2,
+					     RV8263C8_BM_ALARM_INT_ENABLE,
+					     RV8263C8_BM_ALARM_INT_DISABLE);
 	} else {
-		err = i2c_reg_write_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_SECONDS_ALARM,
-					    RV8263C8_BM_ALARM_DISABLE);
+		err = i2c_reg_update_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_CONTROL_2,
+					     RV8263C8_BM_ALARM_INT_ENABLE,
+					     RV8263C8_BM_ALARM_INT_ENABLE);
```

Comment:
I now realized that enabling alarm interrupts prior to setting the alarm time could lead to an inadvertent alarm event.

Safer procedure for setting the alarm time would be:
1. always clear the AIE and AF flag
2. write the alarm settings
3. set the AIE to 1

Reply:
Okay, I changed it. Now the behaviour of the function should be:

`mask` == 0:
- The `AIE` bit is cleared and alarms are globally disabled
- The `AE_x` bits are set and written to the alarm time registers to disable the time-based alarms
- `timeptr` is ignored completely

`mask` != 0:
- The `AIE` and `AIF` bit is cleared and alarms are globally disabled
- The `AE_x` bits are cleared and together written with the time from `timeptr` to the alarm time registers to enable the time-based alarms
- The `AIE` set and alarms are globally enabled